pkg/quantile/agent.go

```diff
@@ -50,14 +50,30 @@ func (a *Agent) Reset() {
 }
 
 // Insert v into the sketch.
-func (a *Agent) Insert(v float64) {
-	a.Sketch.Basic.Insert(v)
-
-	a.Buf = append(a.Buf, agentConfig.key(v))
-	if len(a.Buf) < agentBufCap {
-		return
+func (a *Agent) Insert(v float64, sampleRate float64) {
+	k := agentConfig.key(v)
+	// bounds enforcement
+	if sampleRate < 0 || sampleRate > 1 {
```

Comment:
Since there's division with this later, should it be?
```suggestion
	if sampleRate <= 0 || sampleRate > 1 {
```